pennylane/data/data_manager/foldermap.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains ``FolderMapView`` for reading the ``foldermap.json`` file in the
+datasets bucket.
+"""
+
+
+import typing
+from collections.abc import Mapping
+from pathlib import PurePosixPath
+from typing import Any, List, Literal, Optional, Tuple, Union
+
+from .params import Description, ParamArg, ParamVal
+
+
+# Type for a dataset path, relative to the foldermap.json file
+class DataPath(PurePosixPath):
+    """Type for Dataset Path, relative to the foldermap.json file."""
+
+    def __repr__(self) -> str:
+        return repr(str(self))
+
+
+class FolderMapView(typing.Mapping[str, Union["FolderMapView", DataPath]]):
+    """Provides a read-only view of the ``foldermap.json`` file in
+    the datasets bucket. The folder map is a nested mapping of
+    dataset parameters to their path, relative to the ``foldermap.json``
+    file.
+
+    A dictionary in the folder map can optionally specify a default
+    paramater using the '__default' key. This view hides that
```

Comment:
What do you mean by this? I can still see `__default` key in the example given below.

Reply:
This was an example for the underlying data. Clarified in the docstring.